packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts

```diff
@@ -590,7 +590,11 @@ export class TaskDefinition extends TaskDefinitionBase {
     if (this.networkMode === NetworkMode.BRIDGE || this.networkMode === NetworkMode.NAT) {
       return EPHEMERAL_PORT_RANGE;
     }
-    return portMapping.protocol === Protocol.UDP ? ec2.Port.udp(portMapping.containerPort) : ec2.Port.tcp(portMapping.containerPort);
+    if (portMapping.containerPort !== undefined) {
+      return portMapping.protocol === Protocol.UDP ? ec2.Port.udp(portMapping.containerPort) : ec2.Port.tcp(portMapping.containerPort);
+    }
+    const [startPort, endPort] = portMapping.containerPortRange!.split('-', 2).map(v => +v);
```

Comment:
>Should be equivalent (?)

Yes it is accordingly to this [Stackoverflow answer](https://stackoverflow.com/questions/17106681/parseint-vs-unary-plus-when-to-use-which/17106702#17106702)

>You should still check for NaNs

Maybe it would be better to validate the string with a regex while constructing the PortMapping object so that we don't have to worry about invalid castings here. WDYT?

>and probably tokens?

I don't have much experience with CDK and with tokens, so I'm not sure I understood which check I should add. Also, why bother for tokens here while the rest of the pre-existing code doesn't do it?

Reply:
> Also, why bother for tokens here while the rest of the pre-existing code doesn't do it?

Because the rest of the code is not parsing a string, but just passing on values. Passing token values around is fine, validating or parsing them is not.

> I don't have much experience with CDK and with tokens, so I'm not sure I understood which check I should add.

```ts
if (Token.isUnresolved(x)) {
  throw new Error(`x may not be a token`);
}
```

Reply:
Understood, thanks! I added the relevant code and its unit tests to the `PortMapping` object where I also now test the string value with a regex to ensure it's valid at construction time.